### PR TITLE
refactor / move navigation-logic to `<Scanner/>`-screen

### PR DESCRIPTION
**Pull request by Arxcis**
## Why?
Because we want to make the separation between UI-logic and business-logic clear. Decisions about where to navigate, when, should reside inside screens (UI-layer).

## Changelog
- Add / new hook `useLocalNavigation()`.
- refac / move navigation()-calls to `<Scanner/>`-screen.
- refac / remove unused `requireTrustedID`-parameter on `pair()`.
- refac / remove dead code in `BankID.tsx`.
- update `podfile.lock` and `project.pbxproj`.

### CODE REVIEW

File: src/hooks/useNavigation.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { useNavigation } from "@react-navigation/core";
+
+export const SCREEN_BANKID = "Bankid";
+export const SCREEN_HOME = "Main";
+
+export function useLocalNavigation() {
```
Comment (aridder) [inline src/hooks/useNavigation.ts:6]: Liker denne

File: src/navigation.tsx [hunk 2]
```diff
@@ -82,7 +83,7 @@ export const Navigation = () => {
     return (
         <Stack.Navigator>
             <Stack.Screen
-                name="Main"
+                name={SCREEN_HOME}
```
Comment (aridder) [inline src/navigation.tsx:86]: 👍

File: src/components/scanner/Scanner.tsx [hunk 3]
```diff
@@ -1,50 +1,59 @@
...
+            try {
+                await pairCached(URI);
+            } catch (err) {
+                console.warn("ERROR: await pairCached(URI): ", err);
                 return;
             }
-            await pair(data, true);
-        } catch (e) {
-            console.error(e);
+            navigateBankID();
```
Comment (aridder) [inline src/components/scanner/Scanner.tsx:40]: 👏

File: src/components/scanner/Scanner.tsx [hunk 4]
```diff
@@ -1,50 +1,59 @@
...
-    const { client, isTest, pair } = useContext(Context);
+    const { navigateBankID, navigateHome } = useLocalNavigation();
+    const { isTest, hasTrustedIdentity, pair, pairCached, client } =
+        useContext(Context);
 
-    async function onRead(data: any) {
-        console.log("onRead", data);
-        if (typeof data !== "string") {
+    async function onInput(maybeURI: any) {
+        if (isTest) {
```
Comment (aridder) [inline src/components/scanner/Scanner.tsx:13]: Fjerne dette kanskje?

**Review by aridder — APPROVED** [review]
Ser veldig bra ut!

File: src/components/scanner/Scanner.tsx (hunk 4 above)
Comment (Arxcis) [reply src/components/scanner/Scanner.tsx:13]: God poeng 👍

Comment (Arxcis) [reply src/components/scanner/Scanner.tsx]: Fixed in f9af341

**Pull request merged**